packages/airnode-deployer/src/utils/logger.ts

```diff
@@ -84,3 +90,21 @@ export function debugMode(mode: boolean) {
 export function inDebugMode() {
   return debugModeFlag;
 }
+
+export function setSecret(secret: string) {
+  secrets.push(secret);
+}
+
+export function replaceSecrets(input: string, secrets: string[]) {
+  let output = input;
+  secrets.forEach((secret) => (output = output.replace(secret, '*'.repeat(secret.length))));
```

Comment:
> Hmm, this is happening for every secret and every line that is logged. But I guess we won't have that many secrets and that many logs so it will affect the performance. It is kinda ugly though

Yeah... I didn't have any other ideas, though. Did you happen to have an alternative in mind?

The other option I thought about is passing down the secrets values to the logger and using a conditional for `replaceSecrets`, but it just ended up being weird since we'd be passing them down so many levels.

Reply:
> Did you happen to have an alternative in mind?

I was thinking about passing the secrets as an additional argument in the log functions and do the replace always only on that one passed line. But I'm not sure that's better. I mean it would perform better but not sure about usability.

Reply:
How about we save the secrets into the `secrets`-array (as currently), but the logger functions will take in an options variable of whether to apply `replaceSecrets` to the text which would still check against the `secrets`-array? Best of both worlds :smile:

Reply:
Yeah, let's go with that, that sounds like a nice compromise :slightly_smiling_face: